app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -19,6 +19,8 @@ class Hawkular::MiddlewareManager < ManageIQ::Providers::MiddlewareManager
     DEFAULT_PORT = 80
     default_value_for :port, DEFAULT_PORT
 
+    has_many :middleware_domains, :foreign_key => :ems_id, :dependent => :destroy
+    has_many :middleware_server_groups, :foreign_key => :ems_id, :dependent => :destroy
```

Comment:
Good question. I am ok with hiding the server groups under the domains. Let's wait for the John's input.